### PR TITLE
Feature/use stream controller

### PR DESCRIPTION
**Pull request by renefloor**
Use a stream controller for better control of errors.

### CODE REVIEW

File: lib/src/cache_manager.dart [hunk 1]
```diff
@@ -104,23 +104,35 @@ abstract class BaseCacheManager {
...
       try {
         var webFile = await webHelper.downloadFile(url, authHeaders: headers);
         if (webFile != null) {
-          yield webFile;
+          streamController.add(webFile);
         }
       } catch (e) {
         if (cacheFile == null) {
-          throw e;
+          streamController.addError(e);
```
Comment (yrom) [inline lib/src/cache_manager.dart]: I think this error should be suppressed if nobody was listening the stream

Comment (yrom) [reply lib/src/cache_manager.dart]: check property `streamController.hasListener`

Comment (renefloor) [reply lib/src/cache_manager.dart]: I tested with or without checking on listeners, but it didn't really have any noticable effects when doing that. What is the problem in your opinion when adding errors to the stream when there are no listeners?

Comment (yrom) [reply lib/src/cache_manager.dart]: I thought it would throw an unhandled exception. That's fine if it wouldn't happen.

**Pull request merged**